org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/preferences/formatter/FormatterModifyDialog.java

```diff
@@ -1578,6 +1578,7 @@ private void createCommentsTree() {
 				.gap()
 				.pref(FormatterMessages.FormatterModifyDialog_comments_pref_enable_javadoc, DefaultCodeFormatterConstants.FORMATTER_COMMENT_FORMAT_JAVADOC_COMMENT)
 				.pref(FormatterMessages.FormatterModifyDialog_comments_pref_enable_block, DefaultCodeFormatterConstants.FORMATTER_COMMENT_FORMAT_BLOCK_COMMENT)
+				.pref(FormatterMessages.FormatterModifyDialog_comments_pref_enable_markdown, DefaultCodeFormatterConstants.FORMATTER_COMMENT_FORMAT_MARKDOWN_COMMENT)
```

Comment:
Now that I think about it, this would fit better one position up (or even above the javadoc?). Markdown is more related to javadoc comment, so putting it between block and line comments seems strange.

Reply:
Markdown as the first item looks OK, especially considering that maybe in the future it will be more widely used than the old Javadoc comment type :)

I don't like the last suggestion as it looks like one item somehow depends on the other and you can't enable markdown formatting without enabling javadoc formatting.

But now I start to wonder about the terminology, because a markdown comment is a kind javadoc comment, right? Maybe we should terms like "Markdown Javadoc" and "Classic Javadoc"?

Reply:
And if we want to be pedantic, we should change all occurrences of "Javadoc" with "JavaDoc" as this is how it's consistently used in official documents like https://docs.oracle.com/en/java/javase/23/javadoc/using-markdown-documentation-comments.html#GUID-27E3BB65-BF86-4F8A-9C6C-A864682BEA84